script/prerender/manage.ts

```diff
@@ -0,0 +1,397 @@
+    books: BOOKS,
+    code: CODE_VERSION,
+    id: RELEASE_ID,
+  }, null, 2));
+
+  writeAssetFile('/rex/config.json', JSON.stringify(config, null, 2));
+}
+
+// These functions begin the stack creation/deletion but do not wait for them to finish,
+// since we have other tasks to do while they are running
+async function createWorkersStack() {
+  // Set the timeoutDate, used also by manage()
+  timeoutDate = new Date(1000 * PRERENDER_TIMEOUT_SECONDS + new Date().getTime());
+
+  console.log(`Prerender timeout set to ${timeoutDate}`);
+  console.log('Started workers stack creation');
+
+  return cfnClient.send(new CreateStackCommand({
+    Parameters: [
+      {
+        ParameterKey: 'BucketName',
+        ParameterValue: BUCKET_NAME,
+      },
+      {
+        ParameterKey: 'BucketRegion',
+        ParameterValue: BUCKET_REGION,
+      },
+      {
+        ParameterKey: 'CodeVersion',
+        ParameterValue: CODE_VERSION,
+      },
+      {
+        ParameterKey: 'PublicUrl',
+        ParameterValue: PUBLIC_URL,
+      },
+      {
+        ParameterKey: 'ReleaseIdSuffix',
+        ParameterValue: RELEASE_ID_SUFFIX,
+      },
+      {
+        ParameterKey: 'ValidUntil',
+        ParameterValue: `${timeoutDate.toISOString().slice(0, -5)}Z`,
+      },
+    ],
+    StackName: WORKERS_STACK_NAME,
+    Tags: [
+      {Key: 'Project', Value: 'Unified'},
+      {Key: 'Application', Value: 'Rex'},
+      {Key: 'Environment', Value: 'shared'},
+      {Key: 'Owner', Value: 'dante'},
+    ],
+    TemplateBody: readFile(path.join(__dirname, 'cfn.yml')),
+  }));
+}
+
+async function deleteWorkersStack() {
+  console.log('Started workers stack deletion');
+
+  return cfnClient.send(new DeleteStackCommand({StackName: WORKERS_STACK_NAME}));
+}
+
+async function prepareAndQueueBook([bookId, {defaultVersion}]: [string, {defaultVersion: string}]) {
+  // Don't have the book title yet at this point
+  console.log(`Loading book ${bookId}@${defaultVersion}`);
+
+  const book = await bookLoader(bookId, defaultVersion);
+
+  console.log(`[${book.title}] Book loaded; preparing pages`);
+
+  const pages = prepareBookPages(book).map((page) => omit('route', page.route));
+  const numBookPages = pages.length;
+  numPages += numBookPages;
+
+  // Wait until the work queue is ready
+  const workQueueUrl = await workQueuePromise;
+
+  console.log(`[${book.title}] Queuing ${numBookPages} book pages in batches of 10`);
+
+  for (let pageIndex = 0; pageIndex < pages.length; pageIndex += 10) {
+    const pageBatch = pages.slice(pageIndex, pageIndex + 10);
+
+    // If the entire request fails, this command will throw and be caught at the end of this file
+    // However, we also need to check if only some of the messages failed
+    const sendMessageBatchResult = await sqsClient.send(new SendMessageBatchCommand({
+      Entries: pageBatch.map((page, batchIndex) => ({
+        Id: batchIndex.toString(),
+        MessageBody: JSON.stringify({ payload: { page }, type: 'page' }),
```

Comment:
why the wrapping page key? is there something wrong with the payload just being the `{state, params}` object?

Reply:
I did it for consistency with the sitemap type which needs them but you are right, it is not necessary so I'll remove it.

Reply:
Done